AdobeStockImage/Model/GetImageList.php

```diff
@@ -40,42 +47,83 @@ class GetImageList implements GetImageListInterface
      */
     private $filterBuilder;
 
+    /**
+     * @var ScopeConfigInterface $config
+     */
+    private $config;
+
     /**
      * GetImageList constructor.
      *
      * @param FilterGroupBuilder $filterGroupBuilder
      * @param GetAssetListInterface $getAssetList
      * @param FilterBuilder $filterBuilder
+     * @param ScopeConfigInterface
      * @param array $defaultFilters
      */
     public function __construct(
         FilterGroupBuilder $filterGroupBuilder,
         GetAssetListInterface $getAssetList,
         FilterBuilder $filterBuilder,
+        ScopeConfigInterface $scopeConfigInterface,
```

Comment:
Please create an `\Magento\AdobeStockImage\Model\Config` model and `\Magento\AdobeStockImageApi\Api\ConfigInterface` to handle image related configuration

Reply:
ok

Reply:
Hi @sivaschenko @Nazar65 

I have an opinion that having such a model `Config`  maybe not so good because if it will handle the imaged related configuration it will become an ordinary helper which is not allowed. if it will store the one method only like today it has inconsistency name Config while being a service class it provides a config value.

I would advise 2 variants:
1. Rename this class according to its role - GetDefaultMediaId with one method execute and keep it under the API namespace
2. Rename this class according to is role - GetDefaultMediaId and place it under the Model\COnfig namespace. With this, we will define the config namespace and will be able to put any new configuration related implementation to it. Also, this covers a case when you consider this class Config as not a service API interface.

Some notes from standards:
`6.4.2.3. Each service interface SHOULD declare a single public method. An interface name SHOULD reflect the task or action to be performed. For example, Magento\InventoryApi\Api\StockSourceLinksDeleteInterface::execute(array $links). The only exception is a Repository API, which MAY be added for convenience and MUST be limited to singular CRUD operations and getList($searchCriteria).`

`6.4.1.1. Service contract interfaces SHOULD be placed in separate API modules, except when an existing module already contains Service Contracts in the Api folder. Other modules will depend on the API module, and implementations could be easily swapped via di.xml. API module names must end with the Api suffix. For example, if a module is named MyModule, its APIs SHOULD be declared in a module named MyModuleApi.
`

What do you think about this?